rotkehlchen/db/dbhandler.py

```diff
@@ -3457,3 +3457,22 @@ def create_db_backup(self) -> Path:
             new_db_path,
         )
         return new_db_path
+
+    def get_associated_locations(self) -> List[Location]:
+        cursor = self.conn.cursor()
+        cursor.execute(
+            'SELECT location FROM trades UNION '
+            'SELECT location FROM asset_movements UNION '
+            'SELECT location FROM ledger_actions UNION '
+            'SELECT location FROM margin_positions UNION '
+            'SELECT location FROM user_credentials UNION '
+            'SELECT location FROM amm_swaps',
+
+        )
+        locations_raw = cursor.fetchall()
+        locations = [Location.deserialize_from_db(loc[0]) for loc in locations_raw]
+        cursor.execute('SELECT COUNT(*) FROM balancer_events')
+        balancer_amount = cursor.fetchone()
+        if balancer_amount is not None and balancer_amount[0] >= 1:
```

Comment:
1. The name `balance_amount` is wrong here. It's not an amount. It's a tuple result from a db query. Just use `db_result`.

2. Can it ever be None? I don't think COUNT(*) can ever return null.